ant-bootstrap/src/cache_store/cache_data_v1.rs

```diff
@@ -51,17 +53,26 @@ impl CacheData {
                     while addrs.len() > max_addrs_per_peer {
                         addrs.pop_front();
                     }
+                    found_existing = true;
                     break;
                 }
             }
 
-            self.peers.push_back((*other_peer, other_addrs.clone()));
+            if !found_existing {
+                self.peers.push_back((*other_peer, other_addrs.clone()));
+            }
 
-            while self.peers.len() > max_peers {
-                self.peers.pop_front();
+            // break if limit reached
+            if self.peers.len() >= max_peers {
+                break;
             }
         }
 
+        // Apply max_peers limit by removing from back (oldest from other)
+        while self.peers.len() > max_peers {
```

Comment:
this is somehow no-use as within the previous loop, the breaking out criteria is `if self.peers.len() >= max_peers {` already.

Also, using `self.peers.pop_back()` (new discovered nodes from other is old) here is conflicting with the previous `self.peers.push_back((*other_peer, other_addrs.clone()));` (new discovered nodes from other is new)